packages/@cdktf/provider-generator/lib/get/generator/provider-generator.ts

```diff
@@ -161,44 +166,59 @@ export class TerraformProviderGenerator {
   }
 
   private emitNamespacedResourceFile(
-    namespace: string,
+    providerName: string,
     resources: ResourceModel[]
   ) {
     const ns = resources[0].namespace;
     const comment = ns?.comment;
-    const name = ns?.name || "";
-    const baseFilePath = resources[0].filePath
-      .split("/")
-      .slice(0, -1)
-      .join("/");
-    const filePath = `${baseFilePath}/${name}.ts`;
-
-    const importStatements = [
-      ...new Set( // make list unique
-        resources.reduce(
-          (carry, resource) => [...carry, ...resource.importStatements],
-          [] as string[]
-        )
-      ),
-    ];
-    this.code.openFile(filePath);
-    this.code.line(`// generated from terraform resource schema`);
-    this.code.line();
-    importStatements.forEach((statement) => this.code.line(statement));
-    this.code.line();
-    this.code.line(`/**`);
-    this.code.line(`* ${comment}`);
-    this.code.line(`*/`);
-    this.code.openBlock(`export namespace ${namespace}`);
+
+    if (!ns?.name) throw new Error("namespace name is missing");
+
+    const generatedFiles = [];
+
+    const lowerCasedNamespace = toSnakeCase(ns.name, "");
```

Comment:
Does this produce weird names like for example `f_sx`, `io_t` and `e_c2`? I encountered such behaviour when debugging (and using `toSnakeCase`).

Maybe it might be better to change `ns.name` to the snake case version [in the source](https://github.com/hashicorp/terraform-cdk/blob/3058eb7e1c397f8fb76a5a72ac48b4b39eb162ab/packages/%40cdktf/provider-generator/lib/get/generator/constants/provider-namespaces.ts#L193)?

Reply:
Yes, good point - updated it in bd82746